ui/src/data_explorer/components/SendToDashboardOverlay.tsx

```diff
@@ -0,0 +1,156 @@
+import React, {PureComponent} from 'react'
+
+import {getNewDashboardCell} from 'src/dashboards/utils/cellGetters'
+
+import {
+  OverlayContainer,
+  OverlayHeading,
+  OverlayBody,
+  MultiSelectDropdown,
+} from 'src/reusable_ui'
+
+import {addDashboardCellAsync} from 'src/dashboards/actions'
+
+import {QueryConfig, Dashboard, Source} from 'src/types'
+
+interface Props {
+  queryConfig: QueryConfig
+  dashboards: Dashboard[]
+  source: Source
+  rawText: string
+  onCancel: () => void
+  addDashboardCell: typeof addDashboardCellAsync
+}
+
+interface State {
+  selectedIDs: string[]
+  hasQuery: boolean
+  name: string
+}
+
+class SendToDashboardOverlay extends PureComponent<Props, State> {
+  constructor(props) {
+    super(props)
+    const {queryConfig} = this.props
+
+    this.state = {
+      selectedIDs: [],
+      hasQuery: queryConfig.fields.length !== 0,
+      name: '',
+    }
+  }
+
+  public handleChangeName = e => {
+    const name = e.target.value
+    this.setState({name})
+  }
+
+  public render() {
+    const {onCancel} = this.props
+    const {hasQuery, name} = this.state
+
+    return (
+      <OverlayContainer>
+        <OverlayHeading title="Send to Dashboard" />
+        <OverlayBody>
+          <div className="form-group">
+            <label htmlFor="New Cell Name"> New Cell Name </label>
+            <input
+              type="text"
+              id="New Cell Name"
+              className="form-control input-sm"
+              value={name}
+              onChange={this.handleChangeName}
+            />
+          </div>
+          <MultiSelectDropdown
+            onChange={this.handleSelect}
+            selectedIDs={this.state.selectedIDs}
+          >
+            {this.dropdownItems}
+          </MultiSelectDropdown>
+          <button
+            className="button button-md button-default"
+            onClick={onCancel}
+          >
+            Cancel
+          </button>
+          <button
+            className="button button-md button-success"
+            disabled={!hasQuery}
+            onClick={this.sendToDashboard}
+          >
+            Send to Dashboard
+          </button>
+        </OverlayBody>
+      </OverlayContainer>
+    )
+  }
+
+  private get dropdownItems(): JSX.Element[] {
+    const {dashboards} = this.props
+
+    const simpleArray = dashboards.map(d => ({
```

Comment:
(nitpick) this might read a bit easier if it's done all in one pass
```
    return dashboards.map(dashboard => {
      const id = dashboard.id.toString()

      return (
        <MultiSelectDropdown.Item key={id} id={id} value={id}>
          {dashboard.name}
        </MultiSelectDropdown.Item>
      )
    })
```